Source/JavaScriptCore/bytecode/CallLinkInfo.cpp

```diff
@@ -266,16 +261,13 @@ void CallLinkInfo::revertCallToStub()
     // what in all likelihood fits in 24. So we just splat out the first instruction. Long term, we
     // need something cleaner. But this works on arm64 for now.
 
+    m_callee.clear();
     if (isDataIC()) {
-        m_callee.clear();
         u.dataIC.m_codeBlock = nullptr;
         u.dataIC.m_monomorphicCallDestination = nullptr;
     } else {
 #if ENABLE(JIT)
-        MacroAssembler::repatchPointer(u.codeIC.m_codeBlockLocation, nullptr);
-        CCallHelpers::revertJumpReplacementToBranchPtrWithPatch(CCallHelpers::startOfBranchPtrWithPatchOnRegister(u.codeIC.m_calleeLocation), BaselineJITRegisters::Call::calleeGPR, nullptr);
-#else
-        RELEASE_ASSERT_NOT_REACHED();
```

Comment:
Why does this code (i.e. a non-data IC, but with the JIT disabled) become reachable with this patch?

Reply:
Answer is that the data/non-data ICs are now one and the same